### DISCUSSION TITLE
ahh the dot paragraph is still in my dreams.

### DISCUSSION THREAD

**Discussion by AnthonyMuscio** [wikilabs/plugins#102] [General]
Despite all this wonderful work with custom markup, we must get live, I would still like a leading dot on any line to format a none bolded equivalent of ';" for which I may use the .class method.

Unfortunately I cant do it myself.

```
a line
.this line or paragraph
.this other line or paragraph
another line
```

to appear as
```
a line
this line or paragraph
this other line or paragraph
another line
```

as opposed to 
```
a line
this line or paragraph
this other line or paragraph
another line
```
appearing as

`a line this line or paragraph this other line or paragraph another line`

**Comment by AnthonyMuscio** [comment]
Its like, but not;
```
:not indented
;not bold
*not indented and no bullet
#not indented and no number
!not made larger
```

Basically there are plenty of keyboard entered line based wiki markup symbols, but absolutely no markup that simple treats it as a line. we must always `\n\n` or `<br>` it is so simple and bugged me since 5.1.X

Tony

**Comment by pmario** [comment]
There is absolutely no difference between those 2 code blocks. .. So I'm not sure, what I should see here. 

> to appear as
> 
> ```
> a line
> this line or paragraph
> this other line or paragraph
> another line
> ```
> 
> as opposed to
> 
> ```
> a line
> this line or paragraph
> this other line or paragraph
> another line
> ```

**Reply by AnthonyMuscio** [reply]
Sorry, I just restated the input, sorry.

**Comment by pmario** [comment]
What you want is possible, it just looks a bit different. There are much more possibilities now. 

```
\customize tick _classes=".test-div" 
\customize tick=P _element="p" _classes="test-p" 

´ a line
´P line or paragraph
´.otherClass other line or paragraph
another line
```

will create HTML 

```
<div class="test-div wltc-l1 wltc">a line</div>
<p class="test-p wltc-l1 wltc">line or paragraph</p>
<div class="otherClass test-div wltc-l1 wltc">other line or paragraph</div>
<p>another line</p>
```

 - You have to start using custom-markup at line 1 in a text block. 
   - otherwise the default parser kicks in and ignores custom block rules. 
 - The last line will automatically be covered in a P tag from the standard TW parser

**Reply by AnthonyMuscio** [reply]
Mario, I expected it could already be done, and thanks for spelling it out. 

Why I restated this is when pasting from elsewhere or typing into a tiddler there is no WikiText with which to start a paragraph. It may be as simple as allowing a class to be assigned as with ;.red

I love and value the possibilities with custom mark-up and even support the use of glyphs not available at the keyboard, I can even imagine a small pop-up keyboard to enter such glyphs.

However to me the fact there is no way to introduce a class to the beginning of the, line without needing to close eg @@/@@ and to do so from the keyboard, remains a serious gap. It seems to me there is a hole in wikitext. I am very happy using mouse and button controls except when typing content, if in flow as an author the least resistance the better. So we have wikitext markup which is line based ! !! !! * # ; : etc but all of these come with a predefined behaviour and there is no way for the user to apply a different behaviour using a single character on other lines.

The key barrier seems to be finding a wikitext character to use, but this need not be so difficult if it could be active only on nominated tiddlers. In fact perhaps we could allow **a character to be nominated** for which this occurs. One that comes to mind is & at the beginning of the line, but the reason I liked the . period was that it was not "too visually present" when editing. It would be the users responsibility to comply with the content within the tiddler.

Would it be too difficult to add to the custom wikitext one custom glyph, user nominated so I could go ahead and use period or & to act as ´ in your example? This may be necessary as different keyboards have different additional characters available.

**Reply by AnthonyMuscio** [reply]
Post - is the tick on standard keyboards and I have missed it? the code tick is \` perhaps I could have alt+\` insert a tick glyph?

or as is possible here use the \codetick as I used to escape the meaning of the code tick

**Reply by pmario** [reply]
> Post - is the tick on standard keyboards and I have missed it? the code tick is ` perhaps I could have alt+` insert a tick glyph?
> 
> or as is possible here use the \codetick as I used to escape the meaning of the code tick

I do have a German keyboard, where it is located left of the "backspace" key. https://en.wikipedia.org/wiki/German_keyboard_layout

A good description can be found at: https://superuser.com/questions/254076/how-do-i-type-the-tick-and-backtick-characters-on-windows

**Reply by pmario** [reply]
![grafik](https://user-images.githubusercontent.com/374655/127991648-5fcc57ff-07b7-4be7-839c-aab365d8ec54.png)

 - With CTRL-6 you can open the glyph selector dropdown (1)
 - You can define the glyph you want to use for CTRL-8 there (2).
 - If you select the glyph in (2) the button left to the dropdown will change the icon
 - CTRL-8 will toggle the glyph selected (3)
 - With the configuration (4), you can define some shortcuts (5)
   - So in the screenshot if you hit CTRL-6 and then key-1 you will select "Colorbox Primary", which is predefined
   - The next time you want the same thing just hit CTRL-6  ENTER  because the last command will be remembered in the text input area

**Reply by pmario** [reply]
The same dialogue will be created for the inline-glyphs. TODO ... soon

**Reply by pmario** [reply]
>or as is possible here use the \codetick as I used to escape the meaning of the code tick

It's _not_ the code tick

**Reply by pmario** [reply]
Remember: The glyphs with 1 line are for 1 line-break and 2 line glyphs for 2 line-breaks as the default block-end

**Reply by pmario** [reply]
I think I'll create some videos, once the dialogues are finished.

**Reply by pmario** [reply]
You and @TiddlyTweeter may need to "push" me a bit ;)

**Reply by pmario** [reply]
- So with CTRL-A and CTRL-8 you should be able to define line-glyphs for the whole tiddler .. or
 - Select several lines and hit CTRL-6  key-<something> to add your defined line-prefix

**Comment by pmario** [comment]
The config tiddlers look like this:

![grafik](https://user-images.githubusercontent.com/374655/127994108-49f853a8-63d6-48ce-8086-328991fc1816.png)

**Reply by pmario** [reply]
I just saw, that I may need to change the tag of the Snippet/LinePrefix ...

**Reply by pmario** [reply]
Be aware, that the colour-boxes need 2 linebreaks!

**Reply by AnthonyMuscio** [reply]
I will also revisit this as I have not made use of these features yet.
